test/e2e/volume-provisioner/framework/framework.go

```diff
@@ -315,3 +302,373 @@ func (f *Framework) newConfigurationProvider(cfg *core.Config) (common.Configura
+						Image:           TestContext.Image,
+						Command:         []string{"/usr/local/bin/oci-volume-provisioner"},
+						ImagePullPolicy: v1.PullAlways,
+						Env: []v1.EnvVar{{
+							Name: "NODE_NAME",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "spec.nodeName",
+								},
+							},
+						}, {
+							Name:  "PROVISIONER_TYPE",
+							Value: core.ProvisionerNameDefault,
+						}},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "config",
+							MountPath: "/etc/oci/",
+							ReadOnly:  true,
+						}},
+					},
+					},
+					Volumes: []v1.Volume{{
+						Name: "config",
+						VolumeSource: v1.VolumeSource{
+							Secret: &v1.SecretVolumeSource{
+								SecretName: "oci-volume-provisioner",
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	// TODO(apryde): Decide whether we're adding --controllers="block,filesystem"
+	// to run both provisioners from the same binary and if not dedup this
+	// code. Otherwise it won't be needed :D
+	fssDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "oci-file-system-volume-provisioner",
+			Namespace: "kube-system",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: utilpointer.Int32Ptr(int32(1)),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "oci-file-system-volume-provisioner"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "oci-file-system-volume-provisioner"},
+				},
+				Spec: v1.PodSpec{
+					ServiceAccountName: "oci-volume-provisioner",
+					Containers: []v1.Container{{
+						Name:            "oci-file-system-volume-provisioner",
+						Image:           TestContext.Image,
+						Command:         []string{"/usr/local/bin/oci-volume-provisioner"},
+						ImagePullPolicy: v1.PullAlways,
+						Env: []v1.EnvVar{{
+							Name: "NODE_NAME",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "spec.nodeName",
+								},
+							},
+						}, {
+							Name:  "PROVISIONER_TYPE",
+							Value: core.ProvisionerNameFss,
+						}},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "config",
+							MountPath: "/etc/oci/",
+							ReadOnly:  true,
+						}},
+					},
+					},
+					Volumes: []v1.Volume{{
+						Name: "config",
+						VolumeSource: v1.VolumeSource{
+							Secret: &v1.SecretVolumeSource{
+								SecretName: "oci-volume-provisioner",
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	err := createAndAwaitDeployment(client, blockDeployment)
+	if err != nil {
+		return errors.Wrap(err, "deploying block volume provisioner")
+	}
+
+	err = createAndAwaitDeployment(client, fssDeployment)
+	if err != nil {
+		return errors.Wrap(err, "deploying fss volume provisioner")
+	}
+
+	return nil
+}
+
+func createAndAwaitDeployment(client clientset.Interface, desired *appsv1.Deployment) error {
```

Comment:
Can we move this further up in the framework? It's not specific to the volume provisioner.